File: client/src/app/ipl/types/Match.ts
import { Team } from "./Team";

export class Match {
  matchId: number;
  firstTeamId: number;
  secondTeamId: number;
  firstTeam?: Team;
  secondTeam?: Team;
  matchDate: Date;
  venue: string;
  result: string;
  status: string;
  winnerTeamId: number;
  winnerTeam?: Team;

  constructor(
    matchId: number,
    firstTeamOrId: Team | number,
    secondTeamOrId: Team | number,
    matchDate: Date,
    venue: string,
    result: string,
    status: string,
    winnerTeamOrId: Team | number
  ) {
    this.matchId = matchId;

    if (typeof firstTeamOrId === "number") {
      this.firstTeamId = firstTeamOrId;
    } else {
      this.firstTeam = firstTeamOrId;
      this.firstTeamId = firstTeamOrId.teamId;
    }

    if (typeof secondTeamOrId === "number") {
      this.secondTeamId = secondTeamOrId;
    } else {
      this.secondTeam = secondTeamOrId;
      this.secondTeamId = secondTeamOrId.teamId;
    }

    if (typeof winnerTeamOrId === "number") {
      this.winnerTeamId = winnerTeamOrId;
    } else {
      this.winnerTeam = winnerTeamOrId;
      this.winnerTeamId = winnerTeamOrId.teamId;
    }

    this.matchDate = matchDate;
    this.venue = venue;
    this.result = result;
    this.status = status;
  }

  displayInfo(): void {
    console.log(`Match ID: ${this.matchId}, 
            Match Date: ${this.matchDate}, 
            Venue: ${this.venue}`);
  }
}

File: server/src/main/java/com/edutech/progressive/entity/Match.java
package com.edutech.progressive.entity;

import java.util.Date;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;;

@Entity(name = "matches")
 
public class Match {
 
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int matchId;
 
    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "first_team_id")
    private Team firstTeam;
 
    @ManyToOne(cascade = CascadeType.MERGE)
    @JoinColumn(name = "second_team_id")

    private Team secondTeam;
 
    @Temporal(TemporalType.DATE)
 
    private Date matchDate;
 
    private String venue;
 
    private String result;
 
    private String status;
 
    @ManyToOne
 
    @JoinColumn(name = "winner_team_id")
 
    private Team winnerTeam;
 
    public Match() {
 
    }
 
    public Match(int matchId, int firstTeamId, int secondTeamId, Date matchDate, String venue, String result, String status, int winnerTeamId) {
 
        this.matchId = matchId;
 
        this.firstTeam.setTeamId(firstTeamId);
 
        this.secondTeam.setTeamId(secondTeamId);
 
        this.matchDate = matchDate;
 
        this.venue = venue;
 
        this.result = result;
 
        this.status = status;
 
        this.winnerTeam.setTeamId(winnerTeamId);
 
    }
 
    public int getMatchId() {
 
        return matchId;
 
    }
 
    public void setMatchId(int matchId) {
 
        this.matchId = matchId;
 
    }
 
    public Team getFirstTeam() {
 
        return firstTeam;
 
    }
 
    public void setFirstTeam(Team firstTeam) {
 
        this.firstTeam = firstTeam;
 
    }
 
    public Team getSecondTeam() {
 
        return secondTeam;
 
    }
 
    public void setSecondTeam(Team secondTeam) {
 
        this.secondTeam = secondTeam;
 
    }
 
    public Team getWinnerTeam() {
 
        return winnerTeam;
 
    }
 
    public void setWinnerTeam(Team winnerTeam) {
 
        this.winnerTeam = winnerTeam;
 
    }
 
    public Date getMatchDate() {
 
        return matchDate;
 
    }
 
    public void setMatchDate(Date matchDate) {
 
        this.matchDate = matchDate;
 
    }
 
    public String getVenue() {
 
        return venue;
 
    }
 
    public void setVenue(String venue) {
 
        this.venue = venue;
 
    }
 
    public String getResult() {
 
        return result;
 
    }
 
    public void setResult(String result) {
 
        this.result = result;
 
    }
 
    public String getStatus() {
 
        return status;
 
    }
 
    public void setStatus(String status) {
 
        this.status = status;
 
    }
 
}